docs/decisions/0007-reducing-complexity.md

```diff
@@ -0,0 +1,39 @@
+# Reducing Complexity
+
+## Motivation
+
+High complexity is in general a concerning problem: It can lead to untested legacy code, a high amount of bugs, bad maintainability, slow development, and so on. These decisions are concrete changes we want to make to reduce complexity.
+
+## Decisions
+
+### Hooks
+
+Decision: Hooks always start with `use` (for example `useCounter`). This applies to custom hooks. A custom hook is any function that calls other hooks, especially state and lifecycle hooks like `useEffect` or `useState`. Functions that do not call other hooks never start with `use`.
+
+Reason: See general principles (3). See [React Docs](https://beta.reactjs.org/learn/reusing-logic-with-custom-hooks#hook-names-always-start-with-use)
+
+### Contents of hooks.js
+
+Decision: A hooks.js file lives in the directory of a React component, and must contain only custom hooks. All other functions that are not custom hooks must be in other files. (You may make an exception for functions that you consider private to that hooks.js file and do not expect to be used anywhere else.) All custom hooks that belong to the parent component must be in that hooks.js file. This does not apply for hooks that are intended to be reusable.
```

Comment:
Ditto on design vs. implementation. You're bringing up good points that will improve the quality of our code, and they belong somewhere. I'm just not sure that this is the place. Perhaps the team needs a coding style document, and these go there.

Reply:
Yeah, I don't have a preference where these points should go.

Reply:
I think these points should go here as long as they pertain just concretely to the project (see Adolfo's suggestion below, I am in favor of adding a more general doc or ADR for React best practices. We also have an internal wiki document on React best practices that is basically empty right now.) But I removed any design decisions here except for the point of not having non-hooks in hooks files, and the reason for that is that it is based on community standards and not on a subjective design decision.